docs/proposals/cloud-provider-refactoring.md

```diff
@@ -0,0 +1,106 @@
+## Refactor Cloud Provider out of Kubernetes Core
+
+As kubernetes has evolved tremendously, it has become difficult for different cloudproviders (currently 7) to make changes and iterate quickly. Moreover, the cloudproviders are constrained by the kubernetes build/release lifecycle. This proposal aims to move towards a kubernetes code base where cloud providers specific code will move out of the core repository, and will be maintained by the cloud providers themselves.
+
+### 1. Current use of Cloud Provider
+
+The following components have cloudprovider dependencies
+
+    1. kube-controller-manager
+    2. kubelet
+    3. kube-apiserver
+
+##### Cloud Provider in Kube-Controller-Manager
+The kube-controller-manager has the following controller loops
+
+ - nodeController
+ - volumeController
+ - routeController
+ - serviceController
+ - replicationController
+ - endpointController
+ - resourceQuotaController
+ - namespaceController
+ - deploymentController
+ - etc..
+
+Among these controller loops, the following are cloud provider dependent.
+
+ - nodeController
+ - volumeController
+ - routeController
+ - serviceController
+
+The nodeController uses the cloudprovider to check if a node has been deleted from the cloud. If cloud provider reports a node as deleted, then this controller immediately deletes the node from kubernetes. This check removes the need to wait for a specific amount of time to conclude that an inactive node is actually dead. 
```

Comment:
I think node controller also checks kubelet health.  That part would remain in the core, right?